src/Security/Member.php

```diff
@@ -885,39 +730,37 @@ public static function currentUser()
      */
     public static function actAs($member, $callback)
     {
-        $id = ($member instanceof Member ? $member->ID : $member) ?: 0;
-        $previousID = static::$overrideID;
-        static::$overrideID = $id;
+        $previousUser = Security::getCurrentUser();
+
+        // Transform ID to member
+        if (is_numeric($member)) {
+            $member = DataObject::get_by_id(Member::class, $member);
+        }
+        Security::setCurrentUser($member);
+
         try {
             return $callback();
         } finally {
-            static::$overrideID = $previousID;
+            Security::setCurrentUser($previousUser);
         }
     }
 
     /**
      * Get the ID of the current logged in user
      *
+     * @deprecated use Security::getCurrentUser()
+     *
      * @return int Returns the ID of the current logged in user or 0.
      */
     public static function currentUserID()
     {
-        if (isset(static::$overrideID)) {
-            return static::$overrideID;
-        }
-
-        $id = Session::get("loggedInAs");
-        if (!$id && !self::$_already_tried_to_auto_log_in) {
-            self::autoLogin();
-            $id = Session::get("loggedInAs");
+        if ($member = Security::getCurrentUser()) {
```

Comment:
This method is deprecated, but you are still calling it internally (e.g. from Member:: logged_in_session_exists()). If you deprecate any method make sure you rewrite all usages to avoid deprecated API.

I suggest to deprecate Member::logged_in_session_exists() as well